src/main/webapp/app/iris/state-store.service.ts

```diff
@@ -0,0 +1,148 @@
+
+    constructor() {
+        this.subscription = this.action
+            .pipe(
+                tap((resolvableAction: ResolvableAction) => {
+                    const newState = IrisStateStore.storeReducer(this.state.getValue(), resolvableAction.action);
+                    this.state.next(newState);
+                    if (newState.error) {
+                        resolvableAction.reject(newState.error);
+                    } else {
+                        resolvableAction.resolve();
+                    }
+                }),
+            )
+            .subscribe();
+    }
+
+    ngOnDestroy() {
+        this.subscription.unsubscribe();
+        this.state.complete();
+        this.action.complete();
+    }
+
+    /**
+     * Dispatches an action to update the message store state.
+     * @param action The action to dispatch.
+     * @return promise Promise to handle consecutive cases
+     */
+    dispatchAndThen(action: MessageStoreAction): Promise<void> {
+        return new Promise<void>((resolve, reject) => {
+            this.action.next({
+                action: action,
+                resolve: resolve,
+                reject: reject,
+            });
+        });
+    }
+
+    /**
+     * Dispatches an action to update the message store state.
+     * @param action The action to dispatch.
+     * @return void
+     */
+    dispatch(action: MessageStoreAction): void {
+        this.action.next({
+            action: action,
+            resolve: () => {},
+            reject: () => {},
+        });
+    }
+
+    /**
+     * Returns an observable of the current message store state.
+     * @returns An observable of the current message store state.
+     */
+    getState(): Observable<MessageStoreState> {
+        return this.state.asObservable();
+    }
+
+    /**
+     * Returns an observable of the actions dispatched to the message store.
+     * @returns An observable of the actions dispatched to the message store.
+     */
+    getActionObservable(): Observable<MessageStoreAction> {
+        return this.action.asObservable().pipe(map((resolvableAction: ResolvableAction) => resolvableAction.action));
+    }
+
+    private static exhaustiveCheck(action: never): void {
+        console.debug('You forgot to handle a case of MessageStoreAction: ' + action);
+    }
+
+    private static storeReducer(state: MessageStoreState, action: MessageStoreAction): MessageStoreState {
+        if (state.sessionId == null && !(isSessionReceivedAction(action) || isConversationErrorOccurredAction(action))) {
+            return {
+                messages: [...state.messages],
+                sessionId: state.sessionId,
+                isLoading: false,
+                error: 'Iris ChatBot state is invalid. It is impossible to send messages in such a session.', // TODO translate to German
+            };
+        }
+
+        if (isHistoryMessageLoadedAction(action) || isActiveConversationMessageLoadedAction(action)) {
```

Comment:
How about using a switch case and the default instead of the exhaustiveCheck?

Reply:
We won't gain much by using switch, we will have to handle the default case anyway, and call the exhaustiveCheck method from there

Reply:
Regarding "Unresolved comment": We have another ticket to make error-handling stuff better, and will handle the issue in a separate PR.